third_party/amd/lib/TritonAMDGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -266,6 +267,16 @@ struct ConvertTritonAMDGPUToLLVM
       return signalPassFailure();
     }
 
+    {
+      RewritePatternSet patterns(context);
+      AMD::populateAdjustModeRegisterLLVMPatterns(typeConverter, patterns,
+                                                  targetInfo, AMDBenefit);
+
+      if (failed(applyPatternsGreedily(mod, std::move(patterns)))) {
```

Comment:
This don't need the heavy machinery of patterns which typically is meant for a collection of patterns and keep updating IR until fixed point? I think we just need to walk functions and insert at the beginning directly?